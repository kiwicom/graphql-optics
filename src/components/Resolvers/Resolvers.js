import React from 'react'

const Resolvers = ({ resolvers }) => (
  <table>
    <thead>
      <tr>
        <th>Field name</th>
        <th>Duration</th>
      </tr>
    </thead>
    <tbody>{
      resolvers.map(({ fieldName, duration, path }) => (
        <tr key={fieldName + path.join()}>
          <td>{fieldName}</td>
          <td>{parseInt(duration * 100 / 10 ** 6) / 100} ms</td>
        </tr>
      ))
    }</tbody>
  </table>
)

export default Resolvers
