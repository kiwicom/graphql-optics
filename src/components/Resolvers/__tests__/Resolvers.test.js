import React from 'react'
import Resolvers from '../Resolvers'
import { renderWithRouter } from '../../../__tests__/helpers'

it('renders correctly', () => {
  const data = [
    {fieldName: 'foo', duration: 3242322, path: ['a', 3, 'b']},
    {fieldName: 'bar', duration: 4545, path: ['foo']},
  ]
  const tree = renderWithRouter({ children: <Resolvers resolvers={data} /> })
  expect(tree).toMatchSnapshot()
})
