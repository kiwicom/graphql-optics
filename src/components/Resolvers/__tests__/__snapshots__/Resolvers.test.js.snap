// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<table>
  <thead>
    <tr>
      <th>
        Field name
      </th>
      <th>
        Duration
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        foo
      </td>
      <td>
        3.24
         ms
      </td>
    </tr>
    <tr>
      <td>
        bar
      </td>
      <td>
        0
         ms
      </td>
    </tr>
  </tbody>
</table>
`;
